lib/Index/Index.cpp

```diff
@@ -153,6 +154,27 @@ class IndexSwiftASTWalker : public SourceEntityWalker {
 
   bool IsModuleFile = false;
   bool isSystemModule = false;
+
+  struct ContainerStackEntry {
+    Decl *const visitedDecl;
```

Comment:
The rest of the file generally uses UpperCamelCase for members and locals (current LLVM style). I can see a few places that don't, but it would be good to try to remain consistent (ie. use UpperCamelCase).

Also, you could make this a `const Decl *` if you wanted, but I still don't really see the need for the `* const`.